elements/classpower.lua

```diff
@@ -243,20 +276,30 @@ local function Enable(self, unit)
 		element.ClassPowerEnable = ClassPowerEnable
 		element.ClassPowerDisable = ClassPowerDisable
 
-		local isChildrenTextures
+		local areChildrenTextures, areChildrenStatusBars
 		for i = 1, #element do
 			local icon = element[i]
-			if(icon:IsObjectType('Texture')) then
-				if(not icon:GetTexture()) then
-					icon:SetTexCoord(0.45703125, 0.60546875, 0.44531250, 0.73437500)
-					icon:SetTexture([[Interface\PlayerFrame\Priest-ShadowUI]])
+			if(icon:IsObjectType('StatusBar')) then
+				if(not icon:GetStatusBarTexture()) then
+					icon:SetStatusBarTexture([[Interface\TargetingFrame\UI-StatusBar]])
 				end
 
-				isChildrenTextures = true
+				icon:SetMinMaxValues(0, 1)
+
+				areChildrenStatusBars = true
+			else
+				if(icon:IsObjectType('Texture')) then
+					if(not icon:GetTexture()) then
+						icon:SetTexCoord(0.45703125, 0.60546875, 0.44531250, 0.73437500)
+						icon:SetTexture([[Interface\PlayerFrame\Priest-ShadowUI]])
+					end
+
+					areChildrenTextures = true
+				end
 			end
 		end
```

Comment:
https://github.com/oUF-wow/oUF/issues/346#issuecomment-296351867, yeah :p 

Or do you want to make sure that ALL elements in the table are of the same type? I really hope that no addon dev is retarded enough to through few widget types in the mix.